src/NServiceBus.Core/Unicast/RunningEndpointInstance.cs

```diff
@@ -42,6 +43,7 @@ public RunningEndpointInstance(SettingsHolder settings, IBuilder builder, Pipeli
                 await pipelineCollection.Stop().ConfigureAwait(false);
                 await featureRunner.Stop(messageSession).ConfigureAwait(false);
                 await startAndStoppablesRunner.Stop(messageSession).ConfigureAwait(false);
+                await settings.Get<TransportInfrastructure>().Stop().ConfigureAwait(false);
```

Comment:
Getting it from the settings seems hacky, how about we pass it in the ctor instead?